providers/shared/flows/components/flow.ftl

```diff
@@ -174,169 +181,186 @@
+    [#-- Determine the component type --]
+    [#local component = getComponent(link.Tier, link.Component) ]
+    [#local componentType = getComponentType(component) ]
+
+    [#-- Check if suboccurrence linking is required --]
+    [#-- The preferred approach is to link based on the SubComponent attribute --]
+    [#-- and resolve any same named SubComponents via the Type attribute       --]
+
+    [#local subComponentId = link.SubComponent!"" ]
+    [#if ! subComponentId?has_content]
+        [#-- Support legacy links where SubComponent specific link attributes were used --]
+        [#local subComponents = getComponentChildren(componentType) ]
+        [#list subComponents as subComponent]
+            [#list getComponentChildLinkAttributes(subComponent) as linkAttribute]
+                [#local subComponentId = link[linkAttribute]!"" ]
+                [#if subComponentId?has_content ]
+                    [#if linkType?has_content && (linkType != subComponent.Type) ]
+                        [@fatal
+                            message='Link type of "${linkType}" is inconsistent with SubComponent link attribute "${linkAttribute}"'
+                            context=link
+                        /]
+                        [#return {} ]
+                    [#else]
+                        [#local linkType = subComponent.Type ]
+                        [#local effectiveLink = removeObjectAttributes(effectiveLink, linkAttribute) ]
+                    [/#if]
+                    [#break]
+                [/#if]
+            [/#list]
+            [#if subComponentId?has_content ]
+                [#break]
+            [/#if]
+        [/#list]
+    [/#if]
 
-        [@debug
-            message="Possible link target"
-            context=targetOccurrence
-            enabled=false
-        /]
+    [#-- Update the effective target link --]
+    [#local effectiveLink +=
+        attributeIfContent("SubComponent", subComponentId) +
+        attributeIfContent("Type", linkType)
+    ]
 
-        [#local componentType = getOccurrenceType(targetOccurrence) ]
+    [#-- Get the occurrences matching the effective link    --]
+    [#-- The link is needed to prevent circular references  --]
+    [#-- Because the link is included on the call, only     --]
+    [#-- matches should be been returned                    --]
+    [#-- A match to a component occurrence will also return --]
+    [#-- any SubOccurrences. A match to a SubOccurrence     --]
+    [#-- only returns the Suboccurrence itself              --]
+    [#local targetOccurrences = internalGetOccurrences(component, getTier(link.Tier), effectiveLink) ]
 
-        [#local potentialOccurrences = [targetOccurrence] ]
+    [#if targetOccurrences?size > 1]
+        [@fatal
+            message='Internal error - multiple component matches returned for component "${effectiveLink.Component}"'
+            context=targetOccurrences
```

Comment:
Can we cut this down? Occurrences are huge! not sure if they will help someone figure out an issue

Reply:
I've added a helper routine in occurrence.ftl to abstract exactly what needs to be shown - for now its the type and the core ids to identify the occurrence. We can then tweak the function in future - maybe link it to the logging mode (compact/full) so if you really want the detail, you can get it.